crates/chain/tests/test_local_chain.rs

```diff
@@ -528,6 +528,45 @@ fn checkpoint_from_block_ids() {
     }
 }
 
+#[test]
+fn checkpoint_query() {
+    struct TestCase {
+        chain: LocalChain,
+        /// The heights we want to call [`CheckPoint::query`] with, represented as an inclusive
+        /// range.
+        ///
+        /// If a [`CheckPoint`] exists at that height, we expect [`CheckPoint::query`] to return
+        /// it. If not, [`CheckPoint::query`] should return `None`.
+        query_range: (u32, u32),
+    }
+
+    let test_cases = [
+        TestCase {
+            chain: local_chain![(0, h!("_")), (1, h!("A"))],
+            query_range: (0, 2),
+        },
+        TestCase {
+            chain: local_chain![(0, h!("_")), (2, h!("B")), (3, h!("C"))],
+            query_range: (0, 3),
+        },
+    ];
+
+    for t in test_cases.into_iter() {
+        let tip = t.chain.tip();
+        for h in t.query_range.0..=t.query_range.1 {
+            let query_result = tip.query(h);
+            let exp_hash = t.chain.query(h).map(|cp| cp.hash());
+            match query_result {
+                Some(cp) => {
+                    assert_eq!(Some(cp.hash()), exp_hash);
```

Comment:
It kind of looks like we're getting the checkpoint hash twice here from the same query result.

Reply:
I think this is OK, I read this line as saying: if the `query_result` is some checkpoint, then the hash of that checkpoint must equal the expected hash. 

⛏️ On the other hand I don't see how this test could ever fail since both `tip.query(h)` and `t.chain.query(h)` call `Checkpoint::query()` with `h`.

Reply:
I realized how I wrote this was wrong. We shouldn't be using `query` for both what is to be checked and what we are checking against.

Reply:
> I realized how I wrote this was wrong. We shouldn't be using `query` for both what is to be checked and what we are checking against.

Fixed.